public/components/trace_analytics/components/traces/trace_table_helpers.tsx

```diff
@@ -73,28 +73,44 @@ export const getTableColumns = (
       '-'
     );
 
-  const renderTraceLinkField = (item: string) => (
-    <EuiFlexGroup gutterSize="s" alignItems="center">
-      <EuiFlexItem grow={false}>
-        <EuiLink
-          data-test-subj="trace-link"
-          {...(getTraceViewUri && { href: getTraceViewUri(item) })}
-          {...(openTraceFlyout && { onClick: () => openTraceFlyout(item) })}
-        >
-          <EuiText size="s" className="traces-table traces-table-trace-id" title={item}>
-            {item}
-          </EuiText>
-        </EuiLink>
-      </EuiFlexItem>
-      <EuiFlexItem grow={false}>
-        <EuiCopy textToCopy={item}>
-          {(copy) => (
-            <EuiButtonIcon aria-label="Copy trace id" iconType="copyClipboard" onClick={copy} />
-          )}
-        </EuiCopy>
-      </EuiFlexItem>
-    </EuiFlexGroup>
-  );
+  const renderTraceLinkField = (item: string) => {
+    // Extract the current mode from the URL or session storage
+    const queryParams = new URLSearchParams(window.location.search);
```

Comment:
does this work correctly with hash router?

Reply:
Updated too

```
const currentUrl = window.location.href;
const traceMode = new URLSearchParams(currentUrl.split('?')[1]).get('mode') || sessionStorage.getItem('TraceAnalyticsMode');
```

```
export const appendModeToTraceViewUri = (
  traceId: string,
  getTraceViewUri?: (traceId: string) => string,
  traceMode?: string | null
): string => {
  const baseUri = getTraceViewUri ? getTraceViewUri(traceId) : '';
  const isHashRouter = baseUri.includes('#');
  const separator = isHashRouter ? (baseUri.includes('?') ? '&' : '?') : (baseUri.includes('?') ? '&' : '?');

  if (traceMode) {
    return `${baseUri}${separator}mode=${encodeURIComponent(traceMode)}`;
  }
  return baseUri;
};
```